src/System.IO.FileSystem/src/System/IO/UnixFileStream.cs

```diff
@@ -110,9 +110,17 @@ internal UnixFileStream(String path, FileMode mode, FileAccess access, FileShare
                 _asyncState = new AsyncState();
             }
 
-            // Translate the arguments into arguments for an open call
+            // Translate the arguments into arguments for an open call.
             Interop.Sys.OpenFlags openFlags = PreOpenConfigurationFromOptions(mode, access, options); // FileShare currently ignored
-            Interop.Sys.Permissions openPermissions = Interop.Sys.Permissions.S_IRWXU; // creator has read/write/execute permissions; no permissions for anyone else
+
+            // If the file gets created a new, we'll select the permissions for it.  Most utilities by default use 666 (read and 
+            // write for all). However, on Windows it's possible to write out a file and then execute it.  To maintain that similarity, 
+            // we use 766, so that in additoin the user has execute privileges. No matter what we choose, it'll be subject to the umask 
```

Comment:
NIT: "additoin"  (Sorry, but when I catch a spelling error, I feel obligated to point it out...)